adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+   *                               Dictionary has already been saved to HBase in a previous load operation
+   * @param partitions - number N of partitions in which to repartition data, if not defined data is not repartitioned
+   * @param stagingFolder - user defined location (on HDFS) to which to save temporary staging files
+   */
+
+  def saveVariantContextRDDToHBaseBulk(dao: HBaseSparkDAO,
+                                       vcRdd: VariantContextRDD,
+                                       hbaseTableName: String,
+                                       sequenceDictionaryId: String,
+                                       stagingFolder: String,
+                                       saveSequenceDictionary: Boolean = true,
+                                       partitions: Option[Int] = None) = {
+
+    saveSampleMetadataToHBase(dao, hbaseTableName + "_meta", vcRdd.samples)
+
+    if (saveSequenceDictionary) saveSequenceDictionaryToHBase(dao,
+      hbaseTableName + "_meta",
+      vcRdd.sequences,
+      sequenceDictionaryId)
+
+    val rdd = if (partitions.isDefined) vcRdd.rdd.repartition(partitions.get)
+    else vcRdd.rdd
+
+    val genodata: RDD[(Array[Byte], List[(String, Array[Byte])])] = rdd.mapPartitions((iterator) => {
+      val genotypebaos: java.io.ByteArrayOutputStream = new ByteArrayOutputStream()
+      val genotypeEncoder = EncoderFactory.get().binaryEncoder(genotypebaos, null)
+
+      val genotypeDatumWriter: DatumWriter[Genotype] = new SpecificDatumWriter[Genotype](
+        scala.reflect.classTag[Genotype]
+          .runtimeClass
+          .asInstanceOf[Class[Genotype]])
+
+      val genotypesForHbase: Iterator[(Array[Byte], List[(String, Array[Byte])])] = iterator.map((putRecord) => {
+
+        val myRowKey = KeyStrategy1.getKey(KeyStrategy1rowKeyInfo(putRecord.variant.variant.getContigName,
+          putRecord.variant.variant.getStart.toInt,
+          putRecord.variant.variant.getEnd.toInt,
+          putRecord.variant.variant.getReferenceAllele,
+          putRecord.variant.variant.getAlternateAllele))
+
+        val genotypes: List[(String, Array[Byte])] = putRecord.genotypes.map((geno) => {
+          genotypebaos.reset()
+          genotypeDatumWriter.write(geno, genotypeEncoder)
+          genotypeEncoder.flush()
+          genotypebaos.flush()
+          (geno.getSampleId, genotypebaos.toByteArray)
+        }).toList
+
+        (myRowKey, genotypes)
+      })
+      genotypesForHbase
+    })
+
+    val bulkLoadFlatMap = (t: (Array[Byte], List[(String, Array[Byte])])) => {
+      val data = new ListBuffer[(KeyFamilyQualifier, Array[Byte])]
+      val rowKey = t._1
+      val family = Bytes.toBytes("g")
+      t._2.foreach((x) => {
+        val qualifier: String = x._1
+        val value: Array[Byte] = x._2
+        val mykeyFamilyQualifier = new KeyFamilyQualifier(rowKey, family, Bytes.toBytes(qualifier))
+        data += Tuple2(mykeyFamilyQualifier, value)
```

Comment:
For reason I don't know Scala compiler seems to only accept the explicit Tuple2() which is why I used it.  Maybe some precedence/tokenizing issue  with += and the other nesting.